### PR TITLE
MDEV-14578: mysql_install_db install unix_socket plugin when --auth-r…

### PR DESCRIPTION
**Pull request by grooverdan**
…oot-authentication-method=socket

has minor conflicts with #508

I submit this under the MCA.

### CODE REVIEW

**Comment by grooverdan** [comment]
almost 5 months....

File: sql/sql_plugin.cc [hunk 1]
```diff
@@ -1111,7 +1112,10 @@ static bool plugin_add(MEM_ROOT *tmp_root,
 
   if (name->str && plugin_find_internal(name, MYSQL_ANY_PLUGIN))
   {
-    report_error(report, ER_PLUGIN_INSTALLED, name->str);
+    if (!if_not_exists)
+    {
+      report_error(report, ER_PLUGIN_INSTALLED, name->str);
+    }
     DBUG_RETURN(TRUE);
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply sql/sql_plugin.cc]: bit more complicated, pushing warnings requires a THD. #757

**Comment by grooverdan** [comment]
closing for 10.2 implementation #760

File: sql/sql_plugin.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (vuvova) [reply sql/sql_plugin.cc]: It does, but the caller doesn't need to care. You can push warnings with `my_error`. In fact, you can write to the error log with `my_error` too. Which actually makes a `print_error` function redundant, one can use, say
```
-    report_error(report, ER_CANT_FIND_DL_ENTRY, plugin_interface_version_sym);
+    my_error(ER_CANT_FIND_DL_ENTRY, MYF(report), plugin_interface_version_sym);
```
and `report = 0` or `report = ME_ERROR_LOG` or `report = ME_WARNING`. I'll do it now, this should simplify your #757 a bit.

[the remark this answers is not in the document]
Comment (grooverdan) [reply sql/sql_plugin.cc]: thanks.

**Comment by dbjpanda** [comment]
I still get this issue on 10.3. I am trying to install it on alpine docker `mysql_install_db --datadir=/var/lib/mysql --auth-root-socket-user=dev --auth-root-authentication-method=socket`.

> ERROR: 1126  Can't open shared library '/usr/auth_socket.so' (errno: 2, Error loading shared library /usr/auth_socket.so: No such file or directory)
> 2019-01-09 22:38:37 0 [ERROR] Aborting

**Pull request closed without merging**